pycroft/lib/traffic.py

```diff
@@ -6,188 +6,14 @@
-                           user_id=user.id)
-    session.session.add(credit)
-
-
-@with_transaction
-def grant_regular_credit(user):
-    """Grant a user's regular credit
-
-    The relevant :py:cls:`TrafficGroup` is the one with the largest
-    ``credit_amount``.
-
-    :param User user: the user to grant credit to
-    """
-    now = session.utcnow()
-    group = effective_traffic_group(user)
-    credit = TrafficCredit(timestamp=now, amount=group.credit_amount,
-                           user_id=user.id)
-    session.session.add(credit)
-
-
-@with_transaction
-def reset_credit(user, processor, target_amount=1*1024**3):
-    """Compensate a user's traffic credit to a target amount
-
-    :param User user:
-    :param User processor:
-    :param int target_amount: The target amount to reach, in bytes.
-        Defaults to 1GiB.
-
-    :raises ValueError: if the user's current credit is greater than
-        or equal to the given target amount.
-    """
-    now = session.utcnow()
-    difference = target_amount - user.current_credit
-    if difference <= 0:
-        raise ValueError("The current credit surpasses the target amount."
-                         " Only an upwards correction is possible.")
-
-    session.session.add(TrafficCredit(user=user, timestamp=now, amount=difference))
-    log_user_event(deferred_gettext("Traffic has ben compensated to 1GiB").to_json(),
-                   author=processor,
-                   user=user)
-
-
-@with_transaction
-def sync_exceeded_traffic_limits():
-    """Adds and removes memberships of the 'traffic_limit_exceeded group.'
-
-    """
-
-    processor = User.q.get(0)
-
-    # Add memberships
-    users = User.q.join(User._current_traffic_balance) \
-                  .filter(CurrentTrafficBalance.amount < 0) \
-                  .except_(User.q.join(User.current_properties)\
-                  .filter(or_(CurrentProperty.property_name == 'traffic_limit_disabled',\
-                              CurrentProperty.property_name == 'traffic_limit_exceeded')))\
-                  .all()
-
-    for user in users:
-        make_member_of(user, config.traffic_limit_exceeded_group,
-                       processor, closed(session.utcnow(), None))
-        print("Traffic-Limit exceeded: " + user.name)
-
-    # End memberships
-    users = User.q.join(User.current_properties) \
-                  .filter(CurrentProperty.property_name == 'traffic_limit_exceeded') \
-                  .join(User._current_traffic_balance) \
-                  .filter(or_(CurrentTrafficBalance.amount >= 0,
-                              CurrentProperty.property_name == 'traffic_limit_disabled')) \
-                  .all()
-
-    for user in users:
-        remove_member_of(user, config.traffic_limit_exceeded_group,
-                         processor, closed(session.utcnow(), None))
-        print("Traffic-Limit no longer exceeded: " + user.name)
\ No newline at end of file
+def get_users_with_highest_traffic(limit):
```

Comment:
Did you intend to add this function?

Reply:
Yes, see the referenced issue above (#302).

Reply:
But how is this related to this pull request?

Reply:
Not really. It's a function that we will need when removing the credit system, but it is not directly associated with the PR.

Should I remove it?

Reply:
Making it part of the pull request for #302 would be better in my opinion.

Reply:
I just added the functionality here.